087-single-step-multi-version-downgrade.md

```diff
@@ -0,0 +1,72 @@
+- Downgrading of Kafka where the metadata version in use is greater than the maximum supported by the 'to' Kafka version is not supported
+
+## Motivation
+
+This proposal would make downgrade of Strimzi more user friendly, specifically, in the scenario where a single-step multi version upgrade has been performed and a need arises to downgrade before the metadata version has been advanced. The proposal will allow the user perform the roll back in a single step rather than the multiple steps currently required.
+
+## Proposal
+
+During downgrade an attempt is made to read the Kafka information for the 'from' Kafka version by the 'to' version of the operator. When the 'from' Kafka version is not supported by the 'to' Strimzi version an error will be thrown because the version is unknown. Information such as the metadata version, interbroker protocol message format, log message format version are read from kafka-versions.yaml during the creation of a KafkaVersion object to represent the 'from' kafka version and the error message is generated as it cannot find the information for the unknown version.
+
+However, while this information is important to know for the 'to' version in upgrade and downgrade, and for the 'from' version in upgrade, it is not currently used for the 'from' version in the downgrade. It might be potentially useful in some some future downgrade scenarios, but none that are currently supported.
+
+Kafka does not support downgrade where the metadata version in use is higher than the highest metadata supported by the 'to' Kafka version or where there has been changes in the metadata between the 'from' and 'to' versions. There is already logic in place in Strimzi to prevent such a downgrade from happening. This proposal does not propose any changes to this functionality, it shall continue to be not supported to downgrade in such a scenario. 
+
+### Impacts:
+
+#### Handle unknown versions
+
+KraftVerionChangeCreator.detectToAndFromVersions() invokes the version() method in KafkaVersion.Lookup which throws an exception if the requested version is not known.
+This exception can be caught in detectToAndFromVersions() in the downgrade scenario and an instance of KafkaVersion created that has null values for the unknown data (such as metadata version).
+As this data is not currently read anywhere in the code the null values will not have any negative effect.
```

Comment:
Maybe we should clarify here that we will compare the versions to confirm the assumption that it is a downgrade? (i.e. we will double-check that it is for example 4.5.0 to 4.1.0 and not 3.17.5 to 4.1.0)

Reply:
+1